src/services/vault/client.rs

```diff
@@ -0,0 +1,332 @@
+
+    #[tokio::test]
+    async fn test_verify_token() {
+        let server = MockServer::start();
+
+        let api_token = "secret_token";
+
+        let api_resp = r#"
+            {
+              "data": {
+                "expire_time": "2019-12-10T10:10:10.000000Z",
+                "issue_time": "2019-10-10T10:10:10.000000Z"
+                }
+            }"#;
+
+        let mock_server = server.mock(|when, then| {
+            when.method(GET)
+                .path_contains(VaultClient::VERIFY_TOKEN)
+                .header("X-Vault-Token", api_token);
+            then.status(200)
+                .header("content-type", "application/json; charset=UTF-8")
+                .body(api_resp);
+        });
+
+        let vault_client = VaultClient::new().with_base_url(server.base_url());
+        let response = vault_client.verify_token(api_token).await;
+
+        mock_server.assert();
+        assert!(response.is_ok());
+
+        let response = response.unwrap();
+        assert_eq!(response.status(), 200);
+
+        let verify_token = response.json::<VerifyToken>().await.unwrap();
+
+        assert!(
+            !verify_token.data.expire_time.is_empty(),
+            "Value should not be None"
+        );
+    }
+
+    #[tokio::test]
+    async fn test_fetch_secrets() {
+        let server = MockServer::start();
+
+        let api_token = "test_token";
+        let path = "devenv/test";
+
+        let api_resp = r#"
+            {
+              "data": {
+                "data": {
+                    "test2": "secret_token"
+                    }
+                }
+            }"#;
+
+        let mock_server = server.mock(|when, then| {
+            when.method(GET)
+                .path_contains("/v1/secret/data")
+                .path_contains(path)
+                .header("X-Vault-Token", api_token);
+            then.status(200)
+                .header("content-type", "application/json; charset=UTF-8")
+                .body(api_resp);
+        });
+
+        let vault_client = VaultClient::new().with_base_url(server.base_url());
+        let response = vault_client.fetch_secrets(api_token, path).await;
+
+        mock_server.assert();
+        assert!(response.is_ok());
+
+        let response = response.unwrap();
+        assert_eq!(response.status(), 200);
+
+        let secrets = response.json::<FetchSecrets>().await.unwrap();
+
+        let data = secrets.data.data.get("test2").unwrap();
+        assert_eq!(data, "secret_token");
+    }
+
+    #[tokio::test]
+    async fn test_update_secret() {
+        let server = MockServer::start();
+
+        let api_token = "test_token";
+        let path = "devenv/test";
+        let key = "update_key";
+        let value = "updated_value";
+
+        let api_resp = r#"
+            {
+              "data": {
+                "test2": "secret_token"
+                }
+            }"#;
+
+        let mock_server = server.mock(|when, then| {
+            when.method(PUT)
```

Comment:
I thought we're using `POST` ? Or am i missing something here ?

Reply:
Yeah, missed that update. Updating now